From: Sven Eckelmann <sven@narfation.org>
Date: Fri, 23 May 2025 11:15:53 +0200
Subject: batmand: Add missing string.h include

The build failed because various string functions couldn't be found. This
happened because the indirect include of string.h is now gone in musl
1.2.5.

Signed-off-by: Sven Eckelmann <sven@narfation.org>

--- a/hna.c
+++ b/hna.c
@@ -27,6 +27,7 @@

 #include <errno.h>
 #include <stdlib.h>
+#include <string.h>
 #include <arpa/inet.h>


--- a/linux/kernel.c
+++ b/linux/kernel.c
@@ -29,6 +29,7 @@
 #include <sys/socket.h>
 #include <inttypes.h>
 #include <net/if.h>
+#include <string.h>

 #include "../os.h"
 #include "../batman.h"
--- a/linux/route.c
+++ b/linux/route.c
@@ -29,6 +29,7 @@
 #include <linux/if.h>     /* ifr_if, ifr_tun */
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <string.h>
 #include <sys/socket.h>

 #include "../os.h"
--- a/linux/tun.c
+++ b/linux/tun.c
@@ -32,6 +32,7 @@
 #include <sys/socket.h>
 #include <stdio.h>
 #include <stdlib.h>       /* system() */
+#include <string.h>
 #include <sys/wait.h>     /* WEXITSTATUS */

 #include "../os.h"
